apps/aecore/src/aec_conductor.erl

```diff
@@ -820,22 +871,32 @@ handle_micro_sleep_reply(ok, State) ->
 create_key_block_candidate(#state{keys_ready = false} = State) ->
     %% Keys are needed for creating a candidate
     wait_for_keys(State);
+create_key_block_candidate(#state{key_block_candidates = [{_, #candidate{top_hash = TopHash}} | _],
+                                  top_block_hash       = TopHash} = State) ->
+    %% We have the most recent candidate already. Just start mining.
+    start_mining(State);
 create_key_block_candidate(#state{top_block_hash = TopHash,
                                   beneficiary    = Beneficiary} = State) ->
     epoch_mining:info("Creating key block candidate on the top"),
     Fun = fun() ->
-              {aec_block_key_candidate:create(TopHash, Beneficiary), TopHash}
+                  {aec_block_key_candidate:create(TopHash, Beneficiary), TopHash}
           end,
-    dispatch_worker(create_key_block_candidate, Fun, State).
+    {State1, _Pid} = dispatch_worker(create_key_block_candidate, Fun, State),
```

Comment:
I see you are matching now on `{State, Pid}`, but top branch of case in dispatch_worker returns State?!

Reply:
Well, we should never hit that one, but fixing it anyway... :-P